src/library/scala/collection/mutable/BitSet.scala

```diff
@@ -171,6 +182,113 @@ class BitSet(protected[collection] final var elems: Array[Long])
   override def zip[B](that: IterableOnce[B])(implicit @implicitNotFound(collection.BitSet.zipOrdMsg) ev: Ordering[(Int, B)]): SortedSet[(Int, B)] =
     super.zip(that)
 
+  override def addAll(xs: IterableOnce[Int]): this.type = xs match {
+    case bs: collection.BitSet => this |= bs
+    case range: Range =>
+      if (range.nonEmpty) {
+        val start = range.min
+        if (start >= 0) {
+          val end = range.max
+          val endIdx = end >> LogWL
+          ensureCapacity(endIdx)
+
+          if (range.step == 1 || range.step == -1) {
+            val startIdx = start >> LogWL
+            val wordStart = startIdx * BitSetOps.WordLength
+            val wordMask = -1L << (start - wordStart)
+
+            if (endIdx > startIdx) {
+              elems(startIdx) |= wordMask
+              java.util.Arrays.fill(elems, startIdx + 1, endIdx, -1L)
+              elems(endIdx) |= -1L >>> (BitSetOps.WordLength - (end - endIdx * BitSetOps.WordLength) - 1)
+            } else {
+              elems(endIdx) |= (wordMask & (-1L >>> (BitSetOps.WordLength - (end - wordStart) - 1)))
+            }
+            this
+          } else {
```

Comment:
Could you one-liner-ize all these `else`s?  It's really hard to follow the nesting depth with all the ins-and-outs.  So

```
if (p) {
  stuff
  more.stuff
} else {
  thing
}
```

becomes

```
if (p) {
  stuff
  more.stuff
}
else thing
```

I know that having braces everywhere is said by some to be good practice, but I think it really interferes with code clarity here.

Reply:
I think maybe for pure code this is more acceptable, but in side-effecting blocks, delimiting with curly braces helps protect against error where a second statement is assumed to be in the else branch but actually is outside.